crypto/zk/prover/prover.go

```diff
@@ -0,0 +1,110 @@
+// prover package abstracts the logic and types of go-rapidsnark
+// (https://github.com/iden3/go-rapidsnark) to support basic operations for the
+// rest of vocdoni-node project. It returns custom errors with some more
+// information, trying to clarify the original workflow and standardizing the
+// inputs/outputs types.
+package prover
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/iden3/go-rapidsnark/prover"
+	"github.com/iden3/go-rapidsnark/types"
+	"github.com/iden3/go-rapidsnark/verifier"
+	"github.com/iden3/go-rapidsnark/witness"
+)
+
+// calcWitness perform the witness calculation using go-rapidsnark library based
+// on wasm version of the circuit and inputs provided. To provide the arguments
+// into the correct way, just read the content of wasm binary and inputs JSON
+// files.
+func calcWitness(wasmBytes, inputsBytes []byte) (wtns []byte, err error) {
+	// If the inputs its empty or bad formated, it raise a panic. To avoid it,
+	// catch the panic and return an error instead.
+	defer func() {
+		if perr := recover(); perr != nil {
+			var msg = fmt.Sprintf("error parsing provided circuit inputs, it must be a not empty unmarshalled bytes of a json: %v", perr)
+			err = errors.New(msg)
+		}
+	}()
+
+	// Parse the []byte inputs into a map using the go-rapidsnark/witness
+	// ParseInputs function. Raw JSON Unmarshal result raise an error during
+	// witness calculation.
+	var inputs map[string]interface{}
```

Comment:
normally we declare variables with `foo := bar` instead of using `var foo = bar`